app/src/main/java/org/jellyfin/androidtv/ui/browsing/StdGridFragment.java

```diff
@@ -70,119 +76,324 @@ public class StdGridFragment extends GridFragment {
+    private final double CARD_SPACING_HORIZONTAL_BANNER_PCT = 0.5; // 50% allow horizontal card overlapping for banners, otherwise spacing is too large
+    private final int MIN_GRIDSIZE_CHANGE_DELTA = 4; // minimum pixel size changes, to trigger a recreate of the grid via onGridSizeMeasurements
+    private final int VIEW_SELECT_UPDATE_DELAY = 250; // delay in ms until we update the top-row info for a selected item
+
+    private boolean mDirty = true; // CardHeight, RowDef or GridSize changed
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        if (getActivity() instanceof BaseActivity) mActivity = (BaseActivity) getActivity();
+        backgroundService.getValue().attach(requireActivity());
 
         mFolder = Json.Default.decodeFromString(BaseItemDto.Companion.serializer(), requireActivity().getIntent().getStringExtra(Extras.Folder));
         mParentId = mFolder.getId();
         MainTitle = mFolder.getName();
-        libraryPreferences = preferencesRepository.getValue().getLibraryPreferences(mFolder.getDisplayPreferencesId());
+        libraryPreferences = preferencesRepository.getValue().getLibraryPreferences(Objects.requireNonNull(mFolder.getDisplayPreferencesId()));
         mPosterSizeSetting = libraryPreferences.get(LibraryPreferences.Companion.getPosterSize());
         mImageType = libraryPreferences.get(LibraryPreferences.Companion.getImageType());
         mGridDirection = libraryPreferences.get(LibraryPreferences.Companion.getGridDirection());
+        mCardFocusScale = getResources().getFraction(R.fraction.card_scale_focus, 1, 1);
 
         if (mGridDirection.equals(GridDirection.VERTICAL))
             setGridPresenter(new VerticalGridPresenter());
         else
             setGridPresenter(new HorizontalGridPresenter());
 
-        mCardHeight = getCardHeight(mPosterSizeSetting);
-        setCardHeight(mCardHeight);
+        setDefaultGridRowCols(mPosterSizeSetting, mImageType);
+        setAutoCardGridValues();
+        mJumplistPopup = new JumplistPopup();
+        setupQueries();
+        setupEventListeners();
+    }
 
-        setGridSizes();
+    protected boolean isDirty() {
+        return mDirty;
+    }
 
-        mJumplistPopup = new JumplistPopup();
+    protected BrowseRowDef getRowDef() {
+        return mRowDef;
+    }
+
+    protected void setRowDef(final BrowseRowDef rowDef) {
+        if (mRowDef == null || mRowDef.hashCode() != rowDef.hashCode()) {
+            mDirty = true;
+        }
+        mRowDef = rowDef;
+    }
+
+    protected void setCardHeight(final int height) {
+        if (mCardHeight != height) {
+            mDirty = true;
+        }
+        mCardHeight = height;
+    }
+
+    protected int getCardHeight() {
+        return mCardHeight;
     }
 
-    private void setGridSizes() {
-        Presenter gridPresenter = getGridPresenter();
+    public void printViewStats(View view) {
```

Comment:
Looks unused, can probably be removed

Reply:
its a helper to debug the view, would like to keep this so i don't have to always add it again just to check whats happening with a view.